torch_xla/csrc/ops/cumprod.cpp

```diff
@@ -0,0 +1,58 @@
+#include "torch_xla/csrc/ops/cumprod.h"
+
+#include "torch_xla/csrc/convert_ops.h"
+#include "torch_xla/csrc/helpers.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+#include "torch_xla/csrc/reduction.h"
+#include "torch_xla/csrc/torch_util.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+xla::XlaOp LowerCumProd(const xla::XlaOp& input, xla::int64 dim,
+                        c10::optional<at::ScalarType> dtype) {
+  xla::XlaOp casted_input = CastToScalarType(input, dtype);
+  xla::Shape input_shape = XlaHelpers::ShapeOfXlaOp(casted_input);
+  xla::XlaOp init = XlaHelpers::ScalarValue<float>(
+      1, input_shape.element_type(), casted_input.builder());
+  xla::XlaComputation reducer =
+      XlaHelpers::CreateMulComputation(input_shape.element_type());
+  return BuildCumulativeComputation(casted_input, dim, reducer, init);
+}
+
+xla::Shape NodeOutputShape(const Value& input, xla::int64 dim,
+                           c10::optional<at::ScalarType> dtype) {
+  auto lower_for_shape_fn =
```

Comment:
Do we really have to go through the XLA builder for guessing shapes like these?

Reply:
Done.